iset.mm

```diff
@@ -63822,6 +63846,21 @@ all reals (greatest real number) for which all positive reals are greater.
       KUM $.
   $}
 
+  ${
+    $d C x $.
+    $( Left biconditional closure of disjoint union.  (Contributed by Jim
+       Kingdon, 2-Jul-2022.) $)
+    djulclb $p |- ( C e. V -> ( C e. A <-> ( inl ` C ) e. ( A |_| B ) ) ) $=
```

Comment:
Do we need the sethood antecedent ?
For the right implication, obviously not.
For the left implication, if C were a proper class, then ``(inl ` C)`` would be empty, and the empty set does not belong to a disjoint union, since `( A |_| B ) C_ ( _V X. _V )` (and even `C_ ( 2o X. ( A u. B ) )`, which I think @avekens proved). However, this proof looks non-intuitionistic.

Reply:
> Do we need the sethood antecedent ? For the right implication, obviously not. For the left implication, if C were a proper class, then `` (inl ` C) `` would be empty, and the empty set does not belong to a disjoint union, since `( A |_| B ) C_ ( _V X. _V )` (and even `C_ ( 2o X. ( A u. B ) )`, which I think @avekens proved). However, this proof looks non-intuitionistic.

Most of the reverse closure theorems use excluded middle in their proofs and if there is a way to prove them without it, we haven't found it yet. I don't know if we can make an argument analogous to https://github.com/metamath/set.mm/issues/664#issuecomment-444338796 which might convince ourselves whether it it is even possible in principle.

If any of this gets borrowed into set.mm the situation would be different there, but at least for now I'm only looking at iset.mm.

Reply:
It is easy to show that the empty set cannot be an element of a disjoint union, but to use this for the left implication, we would have to use pm2.61ddc (to get a contradiction if ``C e/ _V`` => ``( inl ` C ) = (/)`` => `` ( ( inl ` C ) e/ ( A |_| B )`` ). But this would require to show ``DECID C e. _V`` (which is not possible in iset.mm?).

Reply:
@avekens : yes, that is the proof I envisioned above, and it is probably non-intuitionistic. Even if we knew it is non-intuitionistic, this would not imply that no intuitionistic proof exists, of course. The comment referenced by @jkingdon above and the following discussion let me think that it is not provable, but I have no precise argument.

As for `DECID C e. _V` being provable, it would imply that `{ C } e. _V` is provable: either C is a set so {C} is a set, or it is not, so {C} is empty, so is a set. And the latter is thought to be not provable (see that same referenced discussion).

Reply:
A simple proof that `DECID C e. _V` is unprovable is that `{ x | ph } e. _V <-> -. ph` (where `ph` does not depend on `x`), so it implies EM for negative propositions.

Reply:
So ("forall C" DECID C e. _V) is equivalent to weak excluded middle. I guess we could define
```
$a |- ( WEXMID <-> A. x ( x C_ 1o -> DECID x = (/) ) $.
```
? And then, `( WEXMID -> DECID -. ph )` and `( WEXMID -> DECID A e. _V )` ?

Reply:
I don't think it's an equivalence, or at least I don't see an obvious way to prove that `C e. _V` is a negative proposition so that weak EM will decide it.

Reply:
Oh yes, you would need to find ph such that C = { x | ph } **where x does not occur in ph**, so `x e. C` will not do...

Reply:
I have put the above result about `C e. _V` into pull request form at https://github.com/metamath/set.mm/pull/2660